include/gridtools/storage/common/storage_info.hpp

```diff
@@ -237,23 +238,23 @@ namespace gridtools {
          * \tparam Dim The index of the dimension
          */
         template <uint_t Dim>
-        GT_FUNCTION GT_CONSTEXPR uint_t padded_length() const {
+        GT_FUNCTION constexpr uint_t padded_length() const {
             return m_padded_lengths[Dim];
         }
 
         /**
          * @brief Returns the array of padded_lengths, the lengths including the halo points (the outer region) and
          * padding.
          */
-        GT_FUNCTION GT_CONSTEXPR const array<uint_t, ndims> &padded_lengths() const { return m_padded_lengths; }
+        GT_FUNCTION constexpr const_ref<array<uint_t, ndims>> padded_lengths() const { return m_padded_lengths; }
```

Comment:
if you use `const_ref` for `padded_lengths`, we should probably also use it for `total_lengths` (at least if they are used in constexpr context).

Reply:
done